spec/booking_spec.rb

```diff
@@ -0,0 +1,151 @@
+require_relative 'spec_helper'
+
+describe "BookingSystem Class" do
+  before do
+    room_num = 1
+    room = Hotel::Room.new(room_num)
+    info = {id: rand(1..300),
+      room: room,
+      start_date: "2018-09-07",
+      end_date: "2018-09-14"
+    }
+
+    reservation = Hotel::Reservation.new(info)
+    @booking = Hotel::BookingSystem.new
+  end
+
+  describe "Initializer" do
+    it "is an instance of BookingSystem" do
+      expect(@booking).must_be_kind_of Hotel::BookingSystem
+      expect(@booking.rooms.length).must_equal 20
+    end
+
+    it "must keep track of rooms" do
+      expect(@booking).must_respond_to :rooms
+      expect(@booking.rooms).must_be_kind_of Array
+      @booking.rooms.each do |room|
+        expect(room).must_be_kind_of Hotel::Room
+      end
+    end
+
+    it "must keep track of reservations" do
+      expect(@booking).must_respond_to :reservations
+      expect(@booking.reservations).must_be_kind_of Array
+      expect(@booking.reservations).must_be :empty?
+
+      @booking.reservations.each do |reservation|
+        expect(reservation).must_be_kind_of Hotel::Reservation
+      end
+    end
+  end
+
+
+  describe "list_reservations" do
+    before do
+      room_num = 2
+      room2 = Hotel::Room.new(room_num)
+      info = {id: 1445,
+        room: room2,
+        start_date: "2018-09-07",
+        end_date: "2018-09-14"
+      }
+
+      reservation2 = Hotel::Reservation.new(info)
+      @booking = Hotel::BookingSystem.new
+    end
+    it "should return all reservations matching the date" do
+      date = "2018-09-11"
+      expect(@booking.list_reservations(date)).must_be_kind_of Array
+      @booking.reservations.each do |reservation|
+        expect(reservation.date_range.check_include(date)).wont_be_nil
+      end
+    end
+
+    it "should not return any reservations that don't match the date" do
+      date = "2018-09-20"
+      expect(@booking.list_reservations(date)).must_be_kind_of Array
+      @booking.reservations.each do |reservation|
+        expect(reservation.date_range.check_include(date)).must_be_nil
+      end
+    end
+
+    it "should return an empty array if no reservations match" do
+      date = "2018-09-20"
+      expect(@booking.list_reservations(date)).must_be_kind_of Array
+      @booking.reservations.each do |reservation|
+        expect(reservation.date_range.check_include(date)).must_be :empty?
+      end
```

Comment:
Above your indicate that `check_include(date)` should return `nil` if if the date isn't included, but here you expect it to return an empty array. It can't be both...